node/connection_base.js

```diff
@@ -113,6 +114,13 @@ TChannelConnectionBase.prototype.onTimeoutCheck = function onTimeoutCheck() {
     }
     if (self.lastTimeoutTime) {
         self.timedOutEvent.emit(self);
+    } else if (!self.remoteName &&
+        self.timers.now() - self.startTime > self.channel.requestDefaults.timeout) {
+        self.timedOutEvent.emit(self, errors.TimeoutError({
```

Comment:
connection only listen on timedOutEvent, so it is the right event to emit. :) 
other issues are fixed.